src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -165,7 +165,7 @@ def run(self, cmd):
                       (returncode, cmd_str, stdout, stderr))
             raise RGWAMCmdRunException(cmd_str, -returncode, stdout, stderr)
 
-        return (stdout, stderr)
+        return (returncode, stdout, stderr)
```

Comment:
could oyu please annotate the function declaration like so?

```python
    def run(self, cmd: str) -> Tuple[int, str, str]:
```

Reply:
Sure, most other functions are not statically typed, so I hadn't noticed that.

Reply:
yeah, we should add those annotations eventually indeed.

Reply:
@sebastian-philipp this change causes make check to fail. Perhaps we should leave the function without typing for now, and then add typing in for every function in a different PR?

```
==================================== ERRORS ====================================
_______________________ ERROR collecting rgw/__init__.py _______________________
rgw/__init__.py:2: in <module>
    from .module import Module
rgw/module.py:14: in <module>
    from ceph.rgw.rgwam_core import EnvArgs, RGWAM
../../python-common/ceph/rgw/rgwam_core.py:139: in <module>
    class RGWCmdBase:
../../python-common/ceph/rgw/rgwam_core.py:155: in RGWCmdBase
    def run(self, cmd: str) -> Tuple[int, str, str]:
E   NameError: name 'Tuple' is not defined
________________________ ERROR collecting rgw/module.py ________________________
rgw/module.py:14: in <module>
    from ceph.rgw.rgwam_core import EnvArgs, RGWAM
../../python-common/ceph/rgw/rgwam_core.py:139: in <module>
    class RGWCmdBase:
../../python-common/ceph/rgw/rgwam_core.py:155: in RGWCmdBase
    def run(self, cmd: str) -> Tuple[int, str, str]:
E   NameError: name 'Tuple' is not defined
```

Reply:
`from typing import Tuple` is enough to fix this!

Reply:
Ok, according to 

https://github.com/ceph/ceph/blob/9c659f228e46591a8edeba85446534a25c260af1/src/mypy.ini#L26-L27

We're of course prone to type errors here.